algorithms/src/Kokkos_StdAlgorithms.hpp

```diff
@@ -48,55 +48,91 @@
 /// \file Kokkos_StdAlgorithms.hpp
 /// \brief Kokkos counterparts for Standard C++ Library algorithms
 
-#include <std_algorithms/Kokkos_Constraints.hpp>
-#include <std_algorithms/Kokkos_RandomAccessIterator.hpp>
+#include <std_algorithms/impl/Kokkos_Constraints.hpp>
+#include <std_algorithms/impl/Kokkos_RandomAccessIterator.hpp>
 #include <std_algorithms/Kokkos_BeginEnd.hpp>
 
 // distance
 #include <std_algorithms/Kokkos_Distance.hpp>
 
-// swap, iter_swap
-#include "std_algorithms/Kokkos_ModifyingOperations.hpp"
+// note that we categorize below the headers
+// following the std classification.
 
-// find, find_if, find_if_not
-// for_each, for_each_n
-// mismatch
-// equal
-// count_if, count
-// all_of, any_of, none_of
-// adjacent_find
-// lexicographical_compare
-// search, search_n
-// find_first_of, find_end
-#include <std_algorithms/Kokkos_NonModifyingSequenceOperations.hpp>
+// modifying ops
+#include "std_algorithms/Kokkos_swap.hpp"
+#include "std_algorithms/Kokkos_iter_swap.hpp"
 
-// replace, replace_copy_if, replace_copy, replace_if
-// copy, copy_n, copy_backward, copy_if
-// fill, fill_n
-// transform
-// generate, generate_n
-// reverse, reverse_copy
-// move, move_backward
-// swap_ranges
-// unique, unique_copy
-// rotate, rotate_copy
-// remove, remove_if, remove_copy, remove_copy_if
-// shift_left, shift_right
-#include <std_algorithms/Kokkos_ModifyingSequenceOperations.hpp>
+// non-modifying sequence
+#include "std_algorithms/Kokkos_adjacent_find.hpp"
+#include "std_algorithms/Kokkos_count.hpp"
+#include "std_algorithms/Kokkos_count_if.hpp"
+#include "std_algorithms/Kokkos_all_of.hpp"
```

Comment:
This naming clashes with the naming convention in the rest of Kokkos.  I am not necessarily saying that this is not negotiable but please motivate that departure.

Reply:
`Kokkos_adjacent_find.hpp` VS `Kokkos_AdjacentFind.hpp

Reply:
Thanks for clarifying. I was using the same name of the algorithm to name the headers. ok, i can fix that to use same naming convention.

Reply:
done